### PR TITLE
Allocate status bar buffer based on sbar height

### PR DESCRIPTION
**Pull request by ceski-1**
See https://github.com/MrAlaux/Nugget-Doom/issues/71

### CODE REVIEW

File: src/st_stuff.c [hunk 1]
```diff
@@ -1038,11 +1039,29 @@ void ST_loadGraphics(void)
...
+      I_Printf(VB_WARNING, "ST_loadGraphics: Non-standard STBAR height of %d. "
+                           "Expected %d.", sbar->height, ST_HEIGHT);
+    }
   }
   else
   {
     sbar  = (patch_t *) W_CacheLumpName("STMBARL", PU_STATIC);
     sbarr = (patch_t *) W_CacheLumpName("STMBARR", PU_STATIC);
+
+    if (sbar && sbar->height != ST_HEIGHT)
```
Comment (fabiangreffrath) [inline src/st_stuff.c:1054]: Not sure if we even need to catch this case. This is strictly only for Doom < 1.2 compatibility.

Comment (ceski-1) [reply src/st_stuff.c]: Fixed by https://github.com/fabiangreffrath/woof/pull/1223/commits/4b284237376c28ff4c8364050fda7f342f90efa2.

**Comment by ceski-1** [comment]
Brief summary:

To prevent warnings,
`STBAR` height must be 32
`STARMS` height must be <= 32
`BRDR_B` height must be <= 32
`STFB*` height must be <= 32

The max height of those lumps is compared to `ST_HEIGHT`. The greater of these two values is used to calculate the buffer size.

**Review by fabiangreffrath — APPROVED** [review]
Looks good, thank you!

**Review by rfomin — APPROVED** [review]

**Pull request merged**